vispy/visuals/volume.py

```diff
@@ -843,8 +843,7 @@ def set_data(self, vol, clim=None, copy=True):
             self._texture.set_clim(clim)
 
         # Apply to texture
-        if should_cast_to_f32(vol.dtype):
-            vol = vol.astype(np.float32)
+        vol = downcast_to_32(vol, copy=copy)
         self._texture.check_data_format(vol)
         self._last_data = vol
         self._texture.scale_and_set_data(vol, copy=copy)  # will be efficient if vol is same shape
```

Comment:
Actually, this should also use `copy` only the first time, for the reasons stated above.